packages/core/src/lib/Cache.ts

```diff
@@ -68,9 +68,8 @@ export class MemoryCache implements ICache {
   }
 
   private startPurgeTimer() {
-    setTimeout(() => {
+    setInterval(() => {
       this.cache.purgeStale()
-      this.startPurgeTimer()
     }, 5000).unref()
```

Comment:
maybe change 5000 to max (2000, min(5000, timeout/2)) ?
if people are feeling the memory pain this much then setting a TTL of 2s will worst case be 10s of memory consumption

Reply:
FWIW I took the 5s number from microsofts memorycache that you're using in equinox

Reply:
I guess every traverse flushed CPU caches so you don't want it to be too frequently
but I also wouldnt read too much into the MS impl coz ultimately its dependent on how the GC interacts wth it
Habing said that, 5000 probably does make sense